File: app/src/main/java/com/simpletodo/aayushf/simpletodo/TagsViewPopulator.java
package com.simpletodo.aayushf.simpletodo;

import android.content.Context;
import android.nfc.Tag;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;
import android.widget.TextView;

import java.util.ArrayList;

/**
 * Created by ayfadia on 5/18/17.
 */

public class TagsViewPopulator extends RecyclerView.Adapter<TagsViewPopulator.Holder> {
    Context c;
    TasksDBHelper h;
    ArrayList<TagItem> t;

    public interface TagsInterface{
        public void tagclicked(String tag);
    }
    public TagsViewPopulator(Context c) {
        super();
        this.c = c;
        this.h = new TasksDBHelper(c);
        Log.d("TAGSVP", "CREATED");
        this.t = h.getTagsList();
        Log.d("TAGSVP", String.valueOf(t.size()));


    }

    @Override
    public Holder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v;
        v = LayoutInflater.from(c).inflate(R.layout.simpledrawerlistitem, null);
        Holder holder = new Holder(v);
        return holder;
    }

    @Override
    public void onBindViewHolder(Holder holder, final int position) {
        holder.tag.setText(t.get(position).tag);
        holder.ptasks.setText(String.valueOf(t.get(position).pendingt));
        Log.d("TAGSVP", String.valueOf(t.size()));
        final TagsInterface i = (TagsInterface)c;
        holder.rl.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                i.tagclicked(t.get(position).tag);

            }
        });
    }


    @Override
    public int getItemCount() {
        return t.size();
    }

    public class Holder extends RecyclerView.ViewHolder{
        TextView tag, ptasks;
        RelativeLayout rl;
        public Holder(View v) {
            super(v);
            tag = (TextView)v.findViewById(R.id.drawerlisttv);
            ptasks = (TextView)v.findViewById(R.id.drawerlistpendingtasks);
            rl = (RelativeLayout)v.findViewById(R.id.rlinsidedraweritem);

        }
    }
}

File: app/src/main/java/com/simpletodo/aayushf/simpletodo/TagsViewPopulator.kt
package com.simpletodo.aayushf.simpletodo

import android.content.Context
import android.nfc.Tag
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.RelativeLayout
import android.widget.TextView

import java.util.ArrayList

/**
 * Created by ayfadia on 5/18/17.
 */

class TagsViewPopulator(internal var c: Context) : RecyclerView.Adapter<TagsViewPopulator.Holder>() {
    internal var h: TasksDBHelper
    internal var t: ArrayList<TagItem>

    interface TagsInterface {
        fun tagclicked(tag: String)
    }

    init {
        this.h = TasksDBHelper(c)
        Log.d("TAGSVP", "CREATED")
        this.t = h.tagsList
        Log.d("TAGSVP", t.size.toString())


    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Holder {
        val v: View
        v = LayoutInflater.from(c).inflate(R.layout.simpledrawerlistitem, null)
        val holder = Holder(v)
        return holder
    }

    override fun onBindViewHolder(holder: Holder, position: Int) {
        holder.tag.text = t[position].tag
        holder.ptasks.text = t[position].pendingt.toString()
        Log.d("TAGSVP", t.size.toString())
        val i = c as TagsInterface
        holder.rl.setOnClickListener { i.tagclicked(t[position].tag) }
    }


    override fun getItemCount(): Int {
        return t.size
    }

    inner class Holder(v: View) : RecyclerView.ViewHolder(v) {
        internal var tag: TextView
        internal var ptasks: TextView
        internal var rl: RelativeLayout

        init {
            tag = v.findViewById(R.id.drawerlisttv) as TextView
            ptasks = v.findViewById(R.id.drawerlistpendingtasks) as TextView
            rl = v.findViewById(R.id.rlinsidedraweritem) as RelativeLayout

        }
    }
}
